### ISSUE TITLE
several domains not parsable

### ISSUE THREAD

**Issue by ckoepp** [joepie91/python-whois#23]
The parser cannot parse the following whois domains:
-   `.co.th` like `starbucks.co.th`
-  `.ir` like `nic.ir`
-  `.com` like `servequake.com`
-  `.com.mx` like `expopack.com.mx` (already referred to issue #22 )

Edit: I'll add non-working tlds with example domains to this post.

**Comment by joepie91** [comment]
All confirmed, will fix soon.

As for `.co.th`, it seems like a switch might be necessary to trigger output in a different format, but I'll have to have a look at this first.

**Comment by ckoepp** [comment]
On 2014-06-24 00:03, Sven Slootweg wrote:

> All confirmed, will fix soon.
> 
> As for `.co.th`, it seems like a switch might be necessary to trigger
> output in a different format, but I'll have to have a look at this
> first.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> 
> https://github.com/joepie91/python-whois/issues/23#issuecomment-46908296

Sure. Take your time!
Unfortunately I'm busy at the university until September so I won't be 
able to submit patches until than :(
But I guess I should be able to find some more non-working tlds.

**Comment by joepie91** [comment]
It looks like .co.th is _very_ inconsistent in its address format... I've found the following variations so far:

```
Domain Holder: Toyota Motor Thailand Co., Ltd. ( บริษัท โตโยต้า มอเตอร์ ประเทศไทย จำกัด )
186/1 Mu1 Old Railway Rd., Samrong Tai, PhraPradaeng, Samut Prakan
10130
TH

Domain Holder: RICOH (THAILAND) CO., LTD. (บริษัท ริโก้ (ประเทศไทย) จำกัด)
Ricoh (Thailand) Ltd.
341 Onnuj Road, 
Prawet, Prawet , 
Bangkok
10250
TH

Tech Contact: 92144
Toyota Motor Asia Pacific Engineering &amp; Manufacturing Co.,Ltd.
99 Moo 5, Bangna-Trad K.M. 29.5 Rd., Ban-Ragad, 
Bang Bor, Samutprakarn
10560
TH

Tech Contact: 491182
Markmonitor
391 N. Ancestor PL 
Boise ID
83704
US

Tech Contact: 503400
บริษัท ริโก้(ประเทศไทย)จำกัด
341 ถนนอ่อนนุช 
แขวงประเวศ  เขตประเวศ
กรุงเทพมหานคร
10250
TH

Tech Contact: 24897
บริษัท เคเอสซี คอมเมอร์เชียล อินเตอร์เนต จำกัด
2/4 อาคารไทยพาณิชย์สามัคคีประกันภัย ชั้น 10 ถนนวิภาวดีรังสิต 
แขวงทุ่งสองห้อง เขตหลักสี่ กรุงเทพฯ
10210
TH

Tech Contact: 15190
991 ศูนย์การค้า สยามพารากอน ถ.พระราม 1 แขวงปทุมวัน เขตปทุมวัน กรุงเทพฯ
10330
TH

Tech Contact: 37581
RS Promotion Public Co.,Ltd.
419/2 Chetchottisak Building JatujakLadphoa Bangkok 10900, bangkok
10900
TH

Domain Holder: Google Inc.
1600 Amphitheatre Parkway
Mountain View, CA 94043
94043
US
```

I'm not sure I can get consistently correct registrant data out of this mess, but I'll try anyway.

**Comment by joepie91** [comment]
I _think_ I have .co.th parsing implemented more-or-less correctly. It turns out that there's no clear delimiter rules in Thai (not even in address data), so I've implemented some very dumb logic that just treats the last 'word' as the city in a number of places. It seems to work so far, but it's not guaranteed to work correctly.

The following is example `pwhois` output:

```
Status            : Active
Registrar         : T.H.NIC Co., Ltd.
Registration date : 2002-03-25 00:00:00
Expiration date   : 2016-03-24 00:00:00
Last update       : 2013-04-12 00:00:00
Name server       : ns.beenets.com
Name server       : ns2.beenets.com

Registrant
    Organization      : Siam Paragon Development co., LTD. (บริษัท สยามพารากอน ดีเวลลอปเม้นท์ จำกัด)
    Street address    : 991 SIAM Paragon Shopping Center RAMA1 ROAD
                        Patumwan
    Postal code       : 10330
    City              : Bangkok
    Country           : TH

Technical Contact
    NIC handle        : 15190
    Street address    : 991 ศูนย์การค้า สยามพารากอน ถ.พระราม 1 แขวงปทุมวัน เขตปทุมวัน
    Postal code       : 10330
    City              : กรุงเทพฯ
    Country           : TH
```

I've been unable to reliably parse district or province information in Thai address listings, so those are only included for Western address listings. In Thai listings, these will likely end up in the 'street address'.

**Comment by joepie91** [comment]
A new version with these fixes (2.3.0) is now available on PyPI. Run `pip install --upgrade pythonwhois` to update. Make sure to read the [update notes](https://github.com/joepie91/python-whois#important-update-notes) prior to updating.
